elyra/pipeline/processor_kfp.py

```diff
@@ -50,6 +50,10 @@ def process(self, pipeline):
         cos_endpoint = runtime_configuration.metadata['cos_endpoint']
         cos_bucket = runtime_configuration.metadata['cos_bucket']
 
+        user_namespace = None
+        if runtime_configuration.metadata['user_namespace']:
```

Comment:
Per my [last comment](https://github.com/elyra-ai/elyra/pull/1081#issuecomment-729879775) I recommend the following:
```suggestion
        if 'user_namespace' in runtime_configuration.metadata and runtime_configuration.metadata['user_namespace']:
```

Reply:
or use `get('user_namespace')`

Reply:
@lresende - removing the if statement and using `get('user_namespace')` when using an empty `user_namespace` field in the runtime metadata returns the original error in the issue. Issue with using `get` atm is that it will never return `None` since the `user_namespace` key will always exist in the metadata dict, even when the field is empty. 
@kevin-bates - Since the key will always exist, the extra check for the key in metadata is redundant. I think the current check should be enough where an empty string would be considered "falsy"

Reply:
I think once the front-end issue is resolved, the key won't exist when no value is present because it won't be persisted.

Reply:
This statement (directly from the docs) should produce an instance in which `user_namespace` is not persisted and is valid metadata:
```
elyra-metadata install runtimes --schema_name=kfp \
       --name=my_kfp \
       --display_name="My Kubeflow Pipeline Runtime" \
       --api_endpoint=https://kubernetes-service.ibm.com/pipeline \
       --api_username=username@email.com \
       --api_password=mypassword \
       --cos_endpoint=http://minio-service.kubeflow:9000 \
       --cos_username=minio \
       --cos_password=minio123 \
       --cos_bucket=test-bucket
```

Reply:
> I think once the front-end issue is resolved, the key won't exist when no value is present because it won't be persisted.

Ive opened #1087 for this. Should we wait then until that issue is fixed and merged before merging this one?

Reply:
There's no need to revisit this if you add the field existence test now.  This does mean a condition will be required before the .get() call.

Reply:
Also note that the condition can occur now via the CLI tool.  You should be able to produce a `KeyError` if not setting a user_namespace value (and without adding the field existence test).